core/src/test/scala/org/apache/spark/util/TimeStampedHashMapSuite.scala

```diff
@@ -118,18 +118,18 @@ class TimeStampedHashMapSuite extends SparkFunSuite {
       assert(testMap2.iterator.toSeq.head === (("k1", "v1")))
 
       // +
-      val testMap3 = testMap2 + (("k0", "v0"))
+      val testMap3 = testMap2 ++ Map("k0" -> "v0")
       assert(testMap3.size === 2)
       assert(testMap3.get("k1").isDefined)
       assert(testMap3("k1") === "v1")
       assert(testMap3.get("k0").isDefined)
       assert(testMap3("k0") === "v0")
 
       // -
-      val testMap4 = testMap3 - "k0"
-      assert(testMap4.size === 1)
-      assert(testMap4.get("k1").isDefined)
-      assert(testMap4("k1") === "v1")
+      testMap3.remove("k0")
```

Comment:
Not sure if this modification aligns with the original test objective.

Reply:
I found that `TimeStampedHashMap` is now a utility class that is no longer used by Spark, and only test cases are still using it.

I personally have the following optional suggestions:
1. Since it is `private[spark]` visibility, perhaps we can directly delete `TimeStampedHashMap`.
2. Or we can mark it as deprecated in Spark 4.0, suppress the compilation warnings with @nowarn, and then remove it in Spark 4.1.
3. Refactor `TimeStampedHashMap` into an `immutable` implementation to maintain the current usage.

cc @dongjoon-hyun FYI

Reply:
https://github.com/apache/spark/commit/b18d70870a33a4783c6b3b787bef9b0eec30bce0#diff-77b12178a7036c71135074c6ddf7d659e5a69906264d5e3061087e4352e304ed  introduced this data structure, and most of its use became ineffective after https://github.com/apache/spark/pull/126

Reply:
+1 for (1) `direct deletion`.